lib/std/fmt/parse_float.zig

```diff
@@ -279,36 +268,34 @@ fn parseRepr(s: []const u8, n: *FloatRepr) !ParseResult {
                     i += 1;
                 } else if (c == 'e' or c == 'E') {
                     i += 1;
-                    state = State.ExponentSign;
+                    state = .ExponentSign;
                 } else {
-                    state = State.ExponentSign;
+                    state = .ExponentSign;
                 }
             },
-
-            State.ExponentSign => {
+            .ExponentSign => {
                 if (c == '+') {
                     i += 1;
                 } else if (c == '-') {
                     negative_exp = true;
                     i += 1;
                 }
 
-                state = State.LeadingExponentZeros;
+                state = .LeadingExponentZeros;
             },
-
-            State.LeadingExponentZeros => {
+            .LeadingExponentZeros => {
                 if (c == '0') {
                     i += 1;
                 } else {
-                    state = State.Exponent;
+                    state = .Exponent;
                 }
             },
-
-            State.Exponent => {
-                if (isDigit(c)) {
-                    if (exponent < std.math.maxInt(i32)) {
-                        exponent *= 10;
-                        exponent += @intCast(i32, c - '0');
+            .Exponent => {
+                if (ascii.isDigit(c)) {
+                    // exponent *= 10
+                    // exponent += c - '0'
+                    if (@mulWithOverflow(i32, exponent, 10, &exponent) or @addWithOverflow(i32, exponent, c - '0', &exponent)) {
```

Comment:
Eh, the fix is even simpler:
```zig
                    if (exponent < std.math.maxInt(i32) / 10) {
```

This is a typo in the Zig translation of the [original](https://github.com/grzegorz-kraszewski/stringtofloat/blob/f472dac6e8032fc45af042b9307fa95f3b5a793a/str2dbl.c#L258) C code.

Reply:
I looked at the original code and got confused because i didn't notice the 2^31 was divided by ten.